Graphs/js/main-directed.js

```diff
@@ -827,3 +827,143 @@ function myPageRank() {
         animateNodes(ranks, 1000);
     }
 }
+
+function heldKarp() {
+    stopAnimation();
+    let root;
+    const [wm, id_index] = getWM(cy, false, true);
+
+    if(!isComplete()){
+        alert('Graph not complete. Please set the probability that two nodes are connected to 1.');
+        return;
+    } 
+
+    if (cy.elements(':selected').length >= 1) {
+        root = cy.elements(':selected')[0];
+    } else {
+        root = prompt('Please enter the id of the start node.\n', '1');
+        root = parseInt(root) - 1;
+    }
+
+    clearCyStyle();
+    
+    const nodes = getAllNodes(cy);
+
+    cy.elements(':grabbable').unselect();
+
+    heldKarpHelp(wm, root);
+}
+
+/**
+ *
+ * @param {number[][]} wm weight matrix
+ * @param {number} x starting point
+ */
+function heldKarpHelp(wm, x) {
+    let n = wm.length;
+    var memo = new Map();
+
+    let nums = new Array(n);
```

Comment:
Use const for variables that are never reassigned
```suggestion
    const nums = new Array(n);
```
If you configured prettier and eslint correctly this should be automatically done for you.